yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -337,5 +339,82 @@ object YarnSparkHadoopUtil {
       conf.getInt("spark.executor.instances", targetNumExecutors)
     }
   }
+
+  /**
+   * Obtains token for the Hive metastore. Exceptions are caught and logged
+   */
+  def obtainTokenForHiveMetastore(
```

Comment:
it does; if None is returned then it means that either the settings say "no need for a token" or the operation failed in a way (no hive on classpath) which was considered not a problem. The latest patch will throw an exception on things like hive metastore connectivity errors.